src/groth16/circuit/builder/mod.rs

```diff
@@ -205,6 +476,64 @@ where
             })
     }
 
+    pub fn evaluate_word8(&mut self, word: &Word8) -> u8 {
+        let mut num: u8 = 0;
+        word.iter().enumerate().for_each(|(i, &wire)| {
+            let eval = self.evaluate(wire);
+            if eval == T::zero() {
+            } else if eval == T::one() {
+                num ^= 0b0000_0001 << i;
+            } else {
+                panic!("evaluate_word8: the evaluation of a wireId is neither 0 or 1");
+            }
```

Comment:
Ah I agree, but that causes the rust compiler to panic. Turns out constant functions in match patterns causes a panic during compilation, since it is not (correctly?) implemented yet. Might be fixed in nightly version of compiler.

Reply:
I'll put a Note: explaining that it should be changed when rustc gets fixed.